### ISSUE TITLE
Thread 1: EXC_RESOURCE RESOURCE_TYPE_MEMORY (limit=12 MB, unused=0x0)

### ISSUE THREAD

**Issue by saitbnzl** [OneSignal/OneSignal-Flutter-SDK#122]
**Description:**
I can get notifications with onesignal_flutter 2.0.1. But I can't get rich notifications, most probably due to this error occuring in OneSignalNotificationServiceExtension.

![Screen Shot 2019-08-09 at 11 50 05](https://user-images.githubusercontent.com/11370176/62766820-fc790e80-ba9b-11e9-87d2-8dd684270df3.png)

**Environment**
onesignal_flutter 2.0.1
flutter v1.7.8+4
Deployment target 10.2
Tested on iPhone 7 (IOS 12.4)

**Steps to Reproduce Issue:**
I'm not sure if you can reproduce it. But I've created my flutter project with swift language (-i swift). It adds use_frameworks for Runner target. Because of this, I had to add use_frameworks! to OneSignalNotificationServiceExtension(#53). Whether I create the extension with Swift or Objective-C I get this error while debugging the extension.

Anyone else having this? Any suggestions?

Thanks.

**Comment by rgomezp** [comment]
You could reach out to our support channel at onesignal.com and send us your project so we can try to reproduce on our end

**Comment by saitbnzl** [comment]
I've finally got it working, but in a different way. Mine is a huge project with multiple schemes also heavily Swift dependent. So I was suffering from many issues with CocoaPods. Now I've integrated OneSignalNotificationServiceExtension without CocoaPods as described [here](https://github.com/OneSignal/OneSignal-iOS-SDK/issues/314). Everything works now. So, I think this is an option, if you are suffering from CocoaPods like me.
